lib/SimpleNonlinearSolve/src/lbroyden.jl

```diff
@@ -62,10 +62,11 @@ end
 
     # For scalar problems / if the threshold is larger than problem size just use Broyden
     if x isa Number || length(x) ≤ η
-        return SciMLBase.__solve(
+        sol = SciMLBase.__solve(
             prob, SimpleBroyden(; alg.linesearch), args...;
             abstol, reltol, maxiters, termination_condition, kwargs...
         )
+        return Utils.nonlinear_solution_new_alg(sol, alg)
```

Comment:
won't this lead to type stability issues?

Reply:
no this will fix it. we need to use simplebroyden for the solve process. But we ensure all branches mark the returned alg as simplelbroyden